Alloy/commands/compile/parsers/Alloy.Abstract.Option.js

```diff
@@ -25,7 +25,7 @@ function parse(node, state, args) {
 
 	if (attrName) {
 		if (args.createArgs[attrName]) {
-			code = `${attrVarName} = ${codePush} - 1`;
+			code = `var ${attrVarName} = ${codePush} - 1`;
 		} else {
 			code = `${attrVarName} = undefined; ${codePush}`;
```

Comment:
I'm pretty sure this has to do with strict mode. Without the `var` this will create a new property on the global object. I guess we were not enforcing strict mode when this change was implemented six years ago. Now this throws an error in strict mode.

You should update the other variable assignment below with the same fix. Then we have all cases covered.

Reply:
ah, that makes sense! I was wondering why it would make a difference not having it declared with var. I've added it for the else case too